streams/src/main/java/org/apache/kafka/streams/state/internals/MeteredKeyValueStore.java

```diff
@@ -103,8 +104,13 @@ public void init(final ProcessorContext context,
 
     @SuppressWarnings("unchecked")
     void initStoreSerde(final ProcessorContext context) {
+        final InternalProcessorContext internalProcessorContext = (InternalProcessorContext) context;
+        final String storeName = name();
+        final String changelogTopic = internalProcessorContext.changelogFor(storeName);
         serdes = new StateSerdes<>(
-            ProcessorStateManager.storeChangelogTopic(context.applicationId(), name()),
+            changelogTopic != null ?
+                changelogTopic :
+                ProcessorStateManager.storeChangelogTopic(context.applicationId(), storeName),
```

Comment:
We discussed this offline. When you revise this to branch based on the context type, you might want to follow the same Util strategy as I proposed here: https://github.com/apache/kafka/pull/8927

Reply:
Ack. 
That means, this PR is now blocked on #8927

Reply:
Nevermind. I copied over `ProcessorContextUtils` from #8927 and extended it. There should not be any conflict.

Reply:
BTW very helpful utils and a great idea. Thank you @vvcephei !

Reply:
No problem! I'm glad it proved useful.